pkg/schema/error.go

```diff
@@ -40,47 +62,49 @@ func NewUnexpectedKeyError(found *yamlmeta.MapItem, definition *filepos.Position
 	}
 }
 
+type schemaAssertionError struct {
+	error
+	description string
+	expected    string
+	found       string
+	hints       []string
+}
+
 type invalidSchemaError struct {
-	Message string
-	Found   yamlmeta.Node
-	Hint    string
+	Title    string
+	FileName string
+	Diff     string
```

Comment:
If I'm reading the usage right, I think `Diff` is the original source.... 

```suggestion
	Source   string
```

?

Reply:
Good idea. done!